docs/spec/README.md

```diff
@@ -0,0 +1,52 @@
+# Source Controller Proposal
+
+## Context
+
+The desired state of a cluster is made out of Kubernetes objects, these objects are expressed in `.yaml` format and 
+are applied on the cluster by operators running inside the cluster. An operator's role is to fetch the Kubernetes
+objects, run transformations on them and reconcile the cluster state with the resulting manifest.
+
+For an operator to acquire the resources that make up the desired state it needs to understand the communication 
+protocol and the authentication scheme, verify the authenticity of a source and deal with rate limits and retries.
+In the FluxCD organization there are currently two operators that perform such operations. Both Flux and 
+Helm Operator connect to Git repositories to fetch Kubernetes objects, they need to maintain an up-to-date mirror 
+of one or several repos. Besides Git, Helm Operator needs to connect to Helm repositories hosted on public or 
+private HTTPS servers.
+
+## Motivation
+
+Each Flux or Helm Operator instance maintains its own Git repository mirror even if all of them
+point to the same source. If the Git repository host becomes unavailable, the cluster state will diverge from the last
+know desired state since the operators will stop the reconciliation due to pull errors. 
+
+Decoupling the Kubernetes objects acquisition from the reconciliation process with an in-cluster 
+source manager would make Flux and Helm Operator resilient to outbound connectivity issues and would
+simplify the state machine(s) that these controllers operate.
+
+Managing the source operations in a dedicated controller could allow Flux to compose the desire state of a cluster
+from multiple source.
+Further more the manifests transformation process could be performed by 3rd party tools
+(e.g. kustomize, jk, tanka, cue run by Tekton pipelines or Kubernetes Jobs)
+that would output the final state of the Kubernetes objects into an artifacts repository maintained by the source controller.
```

Comment:
I can see it might be useful to have a source which Flux syncs, where the source is populated ex-git by some other process. But this seems like a bit of a swerve from the stated purpose of the controller, which is to keep an up-to-date record of what's in git -- now it's do that, _and_ be a storehouse for arbitrary tarballs. I think some alternatives need to be considered.

Reply:
I agree, the source controller could just notify other controllers of a source update and is up to them to run the transformations and pipe the result to Flux.

Reply:
What I would do here is keep the idea, but make this less of a specific suggestion (i.e., remove the statement that the source controller will act as an artifact repository).

Reply:
👌